src/transport/Session.cpp

```diff
@@ -71,10 +71,10 @@ System::Clock::Timeout Session::ComputeRoundTripTimeout(System::Clock::Timeout u
         return System::Clock::kZero;
     }
 
-    // Treat us as active for purposes of GetMessageReceiptTimeout(), pass false into GetAckTimeout to
+    // Treat us as active for purposes of GetMessageReceiptTimeout(), pass false into GetAckTimeout and GetMessageReceiptTimeout to
     // indicate we are processing non-initial message since the other side would be responding to our message.
     return GetAckTimeout(false /*isFirstMessageOnExchange*/) + upperlayerProcessingTimeout +
```

Comment:
So hold on.  Now that I am re-reading this... (and I am sorry we keep going back and forth):

1. GetAckTimeout is how long we expect the message from us to them to take, right?
2. GetMessageReceiptTimeout is how long we expect the message from them to us to take.

For the second one, isFirstMessageOnExchange should definitely be false, since it's responding to our message.

But for the first (GetAckTimeout....  isFirstMessageOnExchange needs to be set based on whether we are computing the roundtrip bits for the first message on the exchange or not, no?

I thought this had gotten fixed in https://github.com/project-chip/connectedhomeip/pull/37696#discussion_r1985465438 so what exactly is going on here?

Reply:
> [force-pushed](https://github.com/project-chip/connectedhomeip/compare/c3a4999a123991f547e8bd9bfa08817fc0d833c5..e3fad9b13b5c62f0fa73b014341f2b1810d932a7) the feature/case_lit_timeout branch from [c3a4999](https://github.com/project-chip/connectedhomeip/commit/c3a4999a123991f547e8bd9bfa08817fc0d833c5) to [e3fad9b](https://github.com/project-chip/connectedhomeip/commit/e3fad9b13b5c62f0fa73b014341f2b1810d932a7)

This is where the broken came in.  See https://github.com/project-chip/connectedhomeip/blob/c3a4999/src/transport/Session.h#L279-L280 which has the right signature, but then it went away.

Reply:
i use vscode github UI to checkout the multiple PRs, somehow it create a new branch automatically with same PRs during PR switches..and lose the right change sets....

working on the right branch

Reply:
I locate the right branch, and compared it against the last comments you make when you mark the previous approval, and add the latest change that fix GetMessageReceiptTimeou